code/modules/mob/living/carbon/human/species.dm

```diff
@@ -1892,6 +1893,110 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 				H.adjustBruteLoss(LOW_PRESSURE_DAMAGE * H.physiology.pressure_mod * delta_time)
 				H.throw_alert("pressure", /atom/movable/screen/alert/lowpressure, 2)
 
+/datum/species/proc/handle_gas_interaction(mob/living/carbon/human/humi, datum/gas_mixture/environment, delta_time, times_fired)
+	if((humi?.wear_suit?.clothing_flags & STOPSPRESSUREDAMAGE) && (humi?.head?.clothing_flags & STOPSPRESSUREDAMAGE))
+		return
+
+	for(var/gas_id in environment.gases)
+		var/gas_amount = environment.gases[gas_id][MOLES]
+		switch(gas_id)
+			if(/datum/gas/antinoblium)
+				if(gas_amount > 30)
+					humi.apply_status_effect(/datum/status_effect/corrosion_curse)
+			if(/datum/gas/freon)
+				if(isplasmaman(humi))
+					if(gas_amount > 5 && gas_amount < 10)
+						to_chat(humi, span_warning("[pick("Your bones feels hot.", "You feel hot.")]"))
+					else
+						to_chat(humi, span_warning("[pick("Your bones starts to heat up!", "You feel very hot!")]"))
+						humi.adjustStaminaLoss(- gas_amount * 0.15 * delta_time)
+						humi.adjust_bodytemperature(max(gas_amount * 0.25, BODYTEMP_HEATING_MAX) * delta_time)
+					continue
+				if(gas_amount > 5 && gas_amount < 10)
+					to_chat(humi, span_warning("[pick("Your skin feels cold.", "You feel cold.")]"))
+				else
+					to_chat(humi, span_warning("[pick("Your skin starts to freeze!", "You feel very cold!")]"))
+					var/current_stamina = humi.getStaminaLoss()
+					if(current_stamina > 25)
+						humi.adjustStaminaLoss(gas_amount * 0.15 * delta_time)
+					humi.adjust_bodytemperature(max(- gas_amount * 0.25, BODYTEMP_COOLING_MAX) * delta_time)
+					if(humi.bodytemperature < 265)
+						humi.apply_status_effect(/datum/status_effect/freon)
+			if(/datum/gas/halon)
+				if(gas_amount > 15)
+					humi.apply_status_effect(/datum/status_effect/stoned)
+			if(/datum/gas/healium)
+				if(gas_amount > 5)
+					to_chat(humi, span_notice("You feel better."))
+					humi.adjustBruteLoss(- gas_amount * 0.5 * delta_time)
+					humi.adjustFireLoss(- gas_amount * 0.15 * delta_time)
+					humi.adjustStaminaLoss(- gas_amount * 0.15 * delta_time)
+			if(/datum/gas/helium)
+				if(gas_amount > 15)
+					humi.AddElement(/datum/element/forced_gravity, 0)
+					addtimer(CALLBACK(humi, .proc/_RemoveElement, list(/datum/element/forced_gravity, 0)), 1 SECONDS * delta_time)
+			if(/datum/gas/hypernoblium)
+				if(gas_amount > 5)
+					humi.apply_status_effect(/datum/status_effect/antimagic)
```

Comment:
Why anti magic?

Reply:
i think it makes the gas more misterious with strange properties, since its crystallized form makes the SM, and it's able to stop every reactions, i like the idea of this gas coating the mobs it touches with a thin layer of it that is able to stop magics